components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/model/BuildTimeMetrics.java

```diff
@@ -0,0 +1,47 @@
+package com.gradle.enterprise.model;
+
+import java.time.Duration;
+
+public class BuildTimeMetrics {
+
+    public final Duration initialBuildTime;
+    public final Duration instantSavings;
+    public final Duration instantSavingsBuildTime;
+    public final Duration pendingSavings;
+    public final Duration pendingSavingsBuildTime;
+
+    private BuildTimeMetrics(
+            Duration initialBuildTime,
+            Duration instantSavings,
+            Duration instantSavingsBuildTime,
+            Duration pendingSavings,
+            Duration pendingSavingsBuildTime) {
+        this.initialBuildTime = initialBuildTime;
+        this.instantSavings = instantSavings;
+        this.instantSavingsBuildTime = instantSavingsBuildTime;
+        this.pendingSavings = pendingSavings;
+        this.pendingSavingsBuildTime = pendingSavingsBuildTime;
+    }
+
+    public static BuildTimeMetrics from(BuildValidationData firstBuild, BuildValidationData secondBuild) {
```

Comment:
I feel this static constructor is too much. Should I condense this all to a single constructor that takes `BuildValidationData, BuildValidationData`?

Reply:
Not sure how you would move everything to the constructor, since you also return null in some cases.

That said, I don't understand the need for these null checks. Non of these values should every be null as per the OpenApi spec.

Reply:
> Not sure how you would move everything to the constructor, since you also return null in some cases.

Good point.

> That said, I don't understand the need for these null checks that then leads to a null object being returned. None of these values should every be null as per the OpenApi spec.

The `null` checks can be removed once we have an alternative [to returning a "default" object](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/c32ab92b4ab9c5201a0c732cb3cf0164e1dbda93/components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/FetchBuildScanDataCommand.java#L87-L112) for a failed Build Scan fetch.

Reply:
I see. Let's not do it in this PR.